common/config/ringpop.go

```diff
@@ -295,6 +300,84 @@ func (provider *dnsProvider) Hosts() ([]string, error) {
 	return results, nil
 }
 
+type dnsSRVProvider struct {
+	UnresolvedHosts []string
+	Resolver        dnsHostResolver
+	Logger          log.Logger
+}
+
+func newDNSSRVProvider(
+	hosts []string,
+	resolver dnsHostResolver,
+	logger log.Logger,
+) *dnsSRVProvider {
+
+	set := map[string]struct{}{}
+	for _, hostport := range hosts {
+		set[hostport] = struct{}{}
+	}
+
+	var keys []string
+	for key := range set {
+		keys = append(keys, key)
+	}
+
+	return &dnsSRVProvider{
+		UnresolvedHosts: keys,
+		Resolver:        resolver,
+		Logger:          logger,
+	}
+}
+
+func (provider *dnsSRVProvider) Hosts() ([]string, error) {
+	var results []string
+	resolvedHosts := map[string][]string{}
+
+	for _, service := range provider.UnresolvedHosts {
+		serviceParts := strings.Split(service, ".")
+		if len(serviceParts) <= 2 {
+			provider.Logger.Warn("could seperate service name from domain", tag.Address(service))
+			continue
+		}
+		serviceName := serviceParts[0]
+		domain := strings.Join(serviceParts[1:], ".")
+		resolved, exists := resolvedHosts[serviceName]
+		if !exists {
+			_, addrs, err := provider.Resolver.LookupSRV(context.Background(), serviceName, "tcp", domain)
+
+			if err != nil {
+				provider.Logger.Warn("could not resolve host", tag.Address(serviceName), tag.Error(err))
```

Comment:
We should error out this case too. Ringpop requires all bootstrap hosts to be discovered, otherwise there can be network partition issue: 
E.g. In a 10 node cluster -- we configure two nodes to be discovered as bootstrap, but only one of them is returned in different times. 
Now it's possible that 5 nodes can discover the same node A, but other 5 nodes discover node B. Then Cadence will start running in a coruppted modes. There are two rings competing each other. They will compete for acquiring Cadence history shards and will constantly having errors.

This is rare, though. Unless you think this error is normal enough that we should ignore.

Reply:
I think that this is a welcome change but it does however change the expected behavior from the DNS method. The previous method just assumes that any non resolvable host is non valid and therefore removed from the true host list. I think your suggestion is expecting DNS to be unavailable for a brief time. I've committed your suggestion.